### PR TITLE
Changes Backpack Slotcount to 21

### PR DESCRIPTION
**Pull request by ghost**
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=4314

Backpacks and all children now have 3 rows of slots instead of just 1. (21 slots total)

However, this is dependent on the size of the bag. For example, satchels and backpacks remain the same, but duffelbags can fit more weight 3 items than backpacks. Each will be able to fit the same amount of cigarettes but duffelbags will fit more laser guns or boxes.

I also changed a message to better show the user what the problem is. If your backpack has space for a small item, but the object you are trying to insert won't fit it will display a different message than 'it's full'.

### CODE REVIEW

**Comment by WJohn** [comment]
So does it handle it by showing duplicate items or something?

Also :+1: for diablo-like inventory, or at least the start.

**Comment by bustygoku** [comment]
Keep in mind subsequent rows of slots do not appear until you've filled every slot in the first row. Nobody will even notice this change and you won't be able to put more guns in your backpack.

**Comment by WJohn** [comment]
Ah, I see. Makes a bit more sense then.

**Comment by Anonus** [comment]
:+1: 
This was something I kind of wanted to do but never bothered trying.

**Comment by bustygoku** [comment]
Go var-edit a backpack to have 21 slots and you'll see for yourself how it works. Pretty simple concept.

**Comment by Anonus** [comment]
For example, you could fit 20 syndicate soaps into a backpack.

**Comment by bustygoku** [comment]
Exactly, but you could already fit 49 soaps in a backpack using boxes if you really wanted to.

**Comment by Miauw62** [comment]
this is a really big buff to backpacks. i hope you can still only put one row of normal-sized items in?

**Comment by WJohn** [comment]
I'm also wondering how big of a buff or nerf this is to bags of holding, can they only hold 21 weight worth of items, but just hold a bigger class per item? Does this mean they could only hold 5 bulky items  and 1 tiny as opposed to the current 7?

**Comment by WJohn** [comment]
Also this REALLY needs a changelog explaining how the mechanics work.

**Comment by Anonus** [comment]
Bags of holding can hold 35 weight of items, so they can hold about as much as they currently do.

**Comment by bustygoku** [comment]
This really isn't a big buff to anything.

Look, yes ,you can put 21 flashbangs in your backpack now.

You can put 21 eswords in your backpack too!

But you can already do that, right now, on the server, with boxes, which are really easy to find. There is pretty much 0 'buffing' in this PR, it's just quality of life so you don't have to stuff everything into boxes.

**Comment by Fox-McCloud** [comment]
There is a huge different between having 21 flashbangs in multiple boxes vs having it in just thebackpack.

The former requires some fiddliness to access the flashbangs, where as the latter is just accessing them on the fly.

This is a pretty huge buff, IMO.

**Comment by bustygoku** [comment]
"Fiddliness" will never be a deterrent to someone running around with a bunch of flashbangs, ever.

**Comment by Fox-McCloud** [comment]
Of course not, but it does force them to waste a few seconds of their time to do so rather than chucking one flashbang after another without consequence.

**Comment by bustygoku** [comment]
Actually grenades are w_class 2 so you can only fit 14 in your backpack

**Comment by Core0verload** [comment]
Still a big buff I think.

**Comment by bustygoku** [comment]
DIdn't you just buff flashbang spam in https://github.com/tgstation/-tg-station/pull/10962 thoguh @core0verload ?

**Comment by Core0verload** [comment]
Flashbang spam is bad and you should feel bad for mentioning it. But I see no significant diffirence between spamming 7 boxes of flashbangs with tetris and without tetris.

**Comment by bustygoku** [comment]
This is a problem with flashbang availability, not storage, is the point I'm getting at here.

Warden gets a box. There's 2(?) in armory. HoS gets a box. Tons are available from the sec vendor. I personally don't care. But leave it out of this PR; flashbang spam will always exist as long as flashbangs in such high quantities exist.

**Comment by Jalleo** [comment]
This is a big bonus to engineering now i can actually carry the boards batteries and other stuff i need for repairs.
If only engineering had more dufflebags.
Unrelated question
Can people make/order any dufflebags?

**Comment by bustygoku** [comment]
Nope, not anywhere else in the code but locker spawns and the syndie ones in the uplink.

**Comment by WJohn** [comment]
Clothing autolathe when

**Comment by Niknakflak** [comment]
I have wanted this for a long time.  There is no reason that 7 crowbars should take the same space as 7 microlasers or some other small object.  Looks good to me.

**Comment by WJohn** [comment]
I think It'd be MUCH more intuitive if we always showed the 3 rows, but had "ghost" items placed on the side for things that are larger than tiny. Right now most people probably won't immediately understand exactly how a bunch of tiny items and a bunch of normal items differ, and exactly where the cap is without counting up each size individually (which'd be a nightmare to do).

**Comment by Lumipharon** [comment]
having 3 rows all the time would clog up half the screen though.

**Comment by bustygoku** [comment]
Someone should add a description line that says 'it is half full' or 'it is full' based on the weight class sum of all the items inside

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -18,6 +18,7 @@
 	slot_flags = SLOT_BACK	//ERROOOOO
 	max_w_class = 3
 	max_combined_w_class = 21
+	storage_slots = 21
```
Comment (freerealestate) [inline code/game/objects/items/weapons/storage/backpack.dm]: I tested this out, you don't actually need this variable anymore.

**Comment by freerealestate** [comment]
you should probably consider removing the slot limit from the smuggler's satchel and the syndicate surgery bag to be consistent, too. they'd both still hold the same amount of normal sized items.

**Comment by Aranclanos** [comment]
this pr is lacking any way to fix the issue of the backpack covering the entire screen because its filled with small items.

**Comment by Anonus** [comment]
That's not an issue, it's a feature.

**Comment by bustygoku** [comment]
@Aranclanos That was already fixed. If you don't put a bunch of tiny items in your backpack, and store them in boxes instead, it won't create the extra rows. The additional backpack rows are only made when you fill the backpack more.

Yeah it's annoying to have 3 rows of backpack slots, but beyond a seperate popup for backpacks that you can minimize (and popups are awful) I don't see another way of doing this. I'd take more slots on my screen and a dynamic backpack weight management than being limited to 7 slots like we are now

**Comment by freerealestate** [comment]
The three rows thing isn't much of an issue at all. The slots are
transparent, so you can see the world behind them. You can use the giant
red X to close the backpack window if it's in the way of something. The
extra rows only appear when they're needed. And if having three rows
bothers you that much, you can just not fill your backpack up with tiny
items.

**Comment by Lumipharon** [comment]
I was directly refering to WJ saying we should always have 3 rows visible.

Also it doesn't matter if you can (partially) see through it - you can't click through it.

**Comment by bustygoku** [comment]
Adding changelog soon.

**Comment by TheVekter** [comment]
Drag and drop inventory management when

**Comment by freerealestate** [comment]
FILO's all we need

**Comment by TheVekter** [comment]
I have OCD, you're triggering me.

**Comment by bustygoku** [comment]
Smuggler satchel and syndicate surgery bag now have the proper slot counts.

Added a changelog.

**Comment by Jalleo** [comment]
:+1: finally my wishes as a engineer are true.

**Pull request merged**